common/ledger/blockledger/fileledger/factory.go

```diff
@@ -56,6 +57,11 @@ func (flf *fileLedgerFactory) ChannelIDs() []string {
 	return channelIDs
 }
 
+// Remove removes block indexes and blocks for the given channelID
+func (flf *fileLedgerFactory) Remove(channelID string) error {
+	return flf.blkstorageProvider.Remove(channelID)
+}
```

Comment:
When is a channelID removed from the `flf.ledgers` ?
Why is this not in sync with `GetOrCreate()`?
I don't see anything that will prevent a user from doing `GetOrCreate()` while this is going on, and getting stale results.